code/obj/item/mob_parts.dm

```diff
@@ -392,3 +396,35 @@
 				if (ispath(streak_splatter))
 					make_cleanable(streak_splatter,src.loc)
 			sleep(0.1 SECONDS)
+
+
+var/global/list/all_valid_random_right_arms = filtered_concrete_typesof(/obj/item/parts, /proc/goes_in_right_arm_slot)
+var/global/list/all_valid_random_left_arms = filtered_concrete_typesof(/obj/item/parts, /proc/goes_in_left_arm_slot)
+var/global/list/all_valid_random_right_legs = filtered_concrete_typesof(/obj/item/parts, /proc/goes_in_right_leg_slot)
+var/global/list/all_valid_random_left_legs = filtered_concrete_typesof(/obj/item/parts, /proc/goes_in_left_leg_slot)
+
+/proc/goes_in_right_arm_slot(var/type)
+	var/obj/item/parts/fakeInstance = type
+	return (((initial(fakeInstance.slot) == "r_arm")) && !(initial(fakeInstance.random_limb_blacklisted)))
+
+/proc/goes_in_left_arm_slot(var/type)
+	var/obj/item/parts/fakeInstance = type
+	return (((initial(fakeInstance.slot) == "l_arm")) && !(initial(fakeInstance.random_limb_blacklisted)))
+
+/proc/goes_in_right_leg_slot(var/type)
+	var/obj/item/parts/fakeInstance = type
+	return (((initial(fakeInstance.slot) == "r_leg")) && !(initial(fakeInstance.random_limb_blacklisted)))
+
+/proc/goes_in_left_leg_slot(var/type)
+	var/obj/item/parts/fakeInstance = type
```

Comment:
Sorry, I think I missed this PR outright and haven't looked through em in a while. The code looks good, but my one critique is here where you do the typecasts for these 4 procs. Since you're not typechecking, and you really don't need to anyway,  the argument to these procs, you can just put the type in the proc definition to avoid the var declaration as the first line in the proc. 

So for each of these, you can change: 
```
/proc/goes_in_left_leg_slot(var/type)
	var/obj/item/parts/fakeInstance = type
```
to: 
```
/proc/goes_in_left_leg_slot(var/obj/item/parts/fakeInstance)
```
Though, that's not really a good name for it once changed. This is really only an aesthetics change to be frank, but I think it's a valid one. It should make it clearer to the reader that this proc should only be given that type as an argument since it can't handle anything else.

Reply:
I followed the format for using filtered_concrete_typesof that was presented in both the [documentation for the proc](https://github.com/goonstation/goonstation/blob/master/_std/types.dm#L88) and in the [coding style guide](https://github.com/goonstation/goonstation/blob/master/.github/CODE_GUIDELINES.md#abstract-types-and-typesof).

Reply:
The reason I bring this up is that your recommendation seems to conflict with the style guide, so uh, I dunno what to follow